extensions/lifecycle/bucketProcessor/LifecycleBucketProcessor.js

```diff
@@ -42,36 +50,46 @@ class LifecycleBucketProcessor {
      * @param {Object} [kafkaConfig.backlogMetrics] - param object to
      * publish kafka topic metrics to zookeeper (see {@link
      * BackbeatConsumer} constructor)
-     * @param {Object} extensions - backbeat config extensions object
-     * @param {Object} extensions.lifecycle - lifecycle config
-     * @param {Object} extensions.lifecycle.auth - authentication info
-     * @param {String} extensions.lifecycle.bucketTasksTopic - lifecycle bucket
+     * @param {Object} lcConfig - lifecycle config
+     * @param {Object} lcConfig.lifecycle.auth - authentication info
+     * @param {String} lcConfig.lifecycle.bucketTasksTopic - lifecycle bucket
      * topic name
-     * @param {Object} extensions.lifecycle.bucketProcessor - kafka consumer
+     * @param {Object} lcConfig.lifecycle.bucketProcessor - kafka consumer
      * object
-     * @param {String} extensions.lifecycle.bucketProcessor.groupId - kafka
+     * @param {String} lcConfig.lifecycle.bucketProcessor.groupId - kafka
      * consumer group id
-     * @param {Object} extensions.replication - replication config
-     * @param {String} extensions.replication.topic - kafka replication topic
-     * @param {Object} extensions.replication.source - replication source
+     * @param {Object} repConfig - replication config
+     * @param {String} repConfig.topic - kafka replication topic
+     * @param {Object} repConfig.source - replication source
      * @param {Number} [lcConfig.bucketProcessor.concurrency] - number
      *  of max allowed concurrent operations
      * @param {Object} s3Config - s3 config
      * @param {String} s3Config.host - host ip
      * @param {String} s3Config.port - port
      * @param {String} transport - http or https
      */
-    constructor(zkConfig, kafkaConfig, extensions, s3Config, transport) {
+    constructor(zkConfig, kafkaConfig, lcConfig, repConfig, s3Config, transport) {
         this._log = new Logger('Backbeat:Lifecycle:BucketProcessor');
         this._zkConfig = zkConfig;
         this._kafkaConfig = kafkaConfig;
-        this._lcConfig = extensions.lifecycle;
-        this._repConfig = extensions.replication;
+        this._lcConfig = lcConfig;
+        this._repConfig = repConfig;
         this._s3Endpoint = `${transport}://${s3Config.host}:${s3Config.port}`;
+        this._s3Config = s3Config;
         this._transport = transport;
         this._producer = null;
         this._kafkaBacklogMetrics = null;
-        this.accountCredsCache = {};
+
+        this._producerReady = false;
+        this._consumerReady = false;
+
+        this.httpAgent = new http.Agent({ keepAlive: true });
+        this.httpsAgent = new https.Agent({ keepAlive: true });
```

Comment:
This might follow a pattern somewhere else but, if you just create one agent here based on the transport we can save some trouble checking it again later on. That is, just create the one agent for the transport we are using and use it everywhere.

Reply:
will be cleaned up when this PR is broken into smaller PRs

Reply:
oh ok cool, I was doing an early review now to make it a little easier but if you are splitting it I will wait.